src/arrayMethodJoin.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joined = '';
+    const formattedSeparator = separator === undefined ? ',' : '' + separator;
```

Comment:
I think Default parameters would be better. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters